e2e/app/modal/share-modal.ts

```diff
@@ -21,33 +20,54 @@ export default class ShareModal extends Modal {
   }
 
   async shareWithUser(userName: string, level: WorkspaceAccessLevel): Promise<void> {
-    const dropDownXpath = `${this.getXpath()}//*[@data-test-id="drop-down"]`;
-    const waitForDropDown = async () => {
-      await waitWhileLoading(this.page);
-      await this.page.waitForXPath(dropDownXpath, { visible: true });
+    const dropDownXpath = this.getXpath() + '//*[@data-test-id="drop-down"][.//clr-icon[@shape="plus-circle"]]';
+    const existsDropDown = async (): Promise<boolean> => {
+      return this.page
+        .waitForXPath(dropDownXpath, { visible: true, timeout: 2000 })
+        .then(() => {
+          return true;
+        })
+        .catch(() => {
+          return false;
+        });
     };
     const waitForClose = async () => {
       await waitWhileLoading(this.page);
       await this.page.waitForXPath(dropDownXpath, { hidden: true });
     };
+    const findCollaboratorAddIcon = () => {
+      return ClrIconLink.findByName(
+        this.page,
+        { type: ElementType.Icon, iconShape: 'plus-circle', containsText: userName, ancestorLevel: 2 },
+        this
```

Comment:
Ideally the search for icon should be done inside the dropdown object. but that change will causes more changes. TODO in a separate coming PR.